pkg/applications/applications_test.go

```diff
@@ -0,0 +1,159 @@
+package applications
+
+import (
+	"testing"
+
+	v1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/apps/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAppendMatchingDeployments(t *testing.T) {
```

Comment:
For what it's worth, I prefer Gingko tests. I think they can be written in a more concise way than Golang table-driven tests. Also, they offer me things like focused tests and better test reports. Probably a matter of taste.